src/Microsoft.ML.Transforms/RandomFourierFeaturizing.cs

```diff
@@ -367,23 +367,15 @@ private float[] GetAvgDistances(RandomFourierFeaturizingEstimator.ColumnInfo[] c
                             for (int i = 0; i < instanceCount; i++)
                             {
                                 for (int j = i + 1; j < instanceCount; j++)
-                                {
-                                    distances[count++] = gaussian ? VectorUtils.L2DistSquared(in res[i], in res[j])
-                                        : VectorUtils.L1Distance(in res[i], in res[j]);
-                                }
+                                    distances[count++] = columns[iinfo].Generator.Distance(in res[i], in res[j]);
```

Comment:
You may want to factor out `columns[iinfo].Generator` into a new variable (here and below). That way we aren't indexing into the array over and over inside these nested for loops.